internal/providers/terraform/azure/mariadb_server.go

```diff
@@ -0,0 +1,112 @@
+package azure
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infracost/infracost/internal/schema"
+
+	"github.com/shopspring/decimal"
+)
+
+func GetAzureMariaDBServerRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_mariadb_server",
+		RFunc: NewAzureMariaDBServer,
+	}
+}
+
+func NewAzureMariaDBServer(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	var costComponents []*schema.CostComponent
+
+	region := d.Get("location").String()
+	sku := d.Get("sku_name").String()
+	tier := strings.Split(sku, "_")[0]
+	family := strings.Split(sku, "_")[1]
+	cores := strings.Split(sku, "_")[2]
```

Comment:
I think we should check the length of the split here so we don't have the possibility of an Index Out of Range error. Maybe we should log if the sku format is not recognised.

Reply:
Are u sure that we need to do this? As Terraform don't give user an opportunity to set wrong sku. 
Example:

```
Running terraform plan

Terraform command failed with:
Error: expected sku_name to be one of [B_Gen5_1 B_Gen5_2 GP_Gen5_2 GP_Gen5_4 GP_Gen5_8 GP_Gen5_16 GP_Gen5_32 MO_Gen5_2 MO_Gen5_4 MO_Gen5_8 MO_Gen5_16], got MO_Gen5
```

Reply:
Nice! It seems they do a lot more validation at the plan stage with Azure. I think it's still best to add this sort of validation on our end. I guess it's pulling those SKUs from the Azure API and we don't know if and when those will change.